app/scripts/lib/validate.js

```diff
@@ -28,28 +44,27 @@ define(function (require, exports, module) {
      * @return true if email is valid, false otw.
      */
     isEmailValid: function (email) {
-      if (typeof email !== 'string') {
+      if (typeof email !== 'string' || email.length > 256) {
         return false;
       }
 
       var parts = email.split('@');
+      var localPart = parts[0];
+      var domainPart = parts[1];
+
+      if (parts.length !== 2 || localPart.length > 64) {
+        return false;
+      }
+
+      if (unicodeRegex.test(localPart)) {
+        localPart = punycode.toASCII(localPart);
```

Comment:
I wrote [some tests](https://github.com/mozilla/fxa-auth-server/compare/phil/issue-1193?expand=1) to check how the auth server behaves and although they pass node 0.10, they fail in 4.3. I'm still looking into why, but it appears to be a bona fide, repeatable failure.

The other thing you mentioned in IRC though, Shane, was Sync. IIUC, even with [bug 1243594](https://bugzilla.mozilla.org/show_bug.cgi?id=1243594) patched in 46+, there is nothing we can do to stop it being a problem for older versions of Firefox. Does that leave unicode email addresses dead in the water for the forseeable future @shane-tomlinson? Should I revert this PR to its pre-punycode incarnation?